src/agent/src/grpc.rs

```diff
@@ -103,20 +104,17 @@ impl agentService {
         // here, the agent will rely on rustjail (using the oci.Mounts
         // list) to bind mount all of them inside the container.
         let m = add_storages(sl!(), req.storages.to_vec(), &self.sandbox)?;
-        {
-            sandbox = self.sandbox.clone();
-            s = sandbox.lock().unwrap();
-            s.container_mounts.insert(cid.clone(), m);
-        }
+        // safe to unwrap() because we don't expect poisoned lock here.
+        let mut s = self.sandbox.lock().unwrap();
+        s.container_mounts.insert(req.container_id.clone(), m);
```

Comment:
We did have some reason to limit the life of lock. rethink about it, otherwise there an be deablock

Reply:
I will recheck the code:)
If needed , we could add a "drop(s)" to solve the possible deadlock.

Reply:
I have checked the original codee again. 'sandbox' and 's'  are declared out side of the block scope, so the lock guard `s` won't be dropped when getting out of the block.
Anything I missed?